### ISSUE TITLE
Dynamic jdbc connection string

### ISSUE THREAD

**Issue by pdex777** [twosigma/beakerx#8141]
Hi!

First of all, thank you for the Beaker project. I've been regularly using it for a few years and I'm now giving a try to the modern BeakerX.

When in SQL language, I create a connection via jdbc using the %defaultDatasource magics. Although the connection itself works fine (i.e. I can retrieve the data from the database), my jdbc connection string contains sensitive information (i.e. login/password) which I would prefer not to save as plain text in the notebook.

So I wonder whether there is away to dynamically construct a jdbc connection string and feed it to the %defaultDatasource magic, by analogy with how the "%classpath add dynamyc" magic works.

Overall, your advice on how to avoid storing credentials in plain for custom jdbc connections in BeakerX is much appreciated.

Thanks!

**Comment by jaroslawmalekcodete** [comment]
Hi @pdex777 
We've have just added new functionality ([PR](https://github.com/twosigma/beakerx/pull/8148) not merged yet to master) to %defaultDatasource magic command that should avoid storing credentials as plain text in the notebook, please look at the [example](https://github.com/twosigma/beakerx/blob/jarek/8141_dynamic_jdbc_connection_string/test/ipynb/sql/SQL_user_password_Test.ipynb) and let us know if it works for you.

In short
1. Configure beakerx.PROPERTY_NAME using groovy kernel
2. Use **{$beakerx.PROPERTY_NAME}** syntax in %defaultDatasource 
```
%defaultDatasource jdbc:postgresql://localhost:5432/postgres?user={$beakerx.dbuser}&password={$beakerx.dbpassword}
```

![DynamiJdbcConnectionString](https://user-images.githubusercontent.com/21952713/67006248-ef544d80-f0e4-11e9-8a27-493e40f4a5f8.png)

**Comment by pdex777** [comment]
Hi @jaroslawmalekcodete ! Thank you for adding this functionality. I would love to give it a try. However, I'm currently using the version of BeakerX from Conda forge (which is 1.4.1 at the time of writing), so this new functionality is not there yet. 

Are there other ways like e.g. daily builds where I can try it out?

Do you have an estimate when the future is going to appear in Conda forge?

Thanks!

**Comment by jaroslawmalekcodete** [comment]
Hi @pdex777 
The PR was merged so you can try to install our master using our instructions https://github.com/twosigma/beakerx#build-and-install-linux-and-mac
